Keep update value prop of NumericInput with observable and action

I-Chenn opened the pull request:
**Description**

The PR is to fix #1906 again with a different idea.

Since the fix in #2104 does not work for all the cases using `SafeNumericInput` thus I create another branch for another method of solving the problem.
In this fix, Decorators `@observable` and `@action` are applied to keep updating the `value` prop of the `NumericInput` used in `SafeNumericInput`.
Please test if there's any problem for those Input columns using the `SafeNumericInput`.

**Checklist**

For linked issues (if there are):
- [X] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [X] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] changelog updated / no changelog update needed
- [X] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [X] `BackendService` unchanged / ~`BackendService` changed and corresponding ICD test fix added~

I-Chenn committed:
Keep update value prop of NumericInput with observable and action

I-Chenn committed:
Take onBlur and onKeyDown into consideration

I-Chenn commented:
In f2a9c6a, the situation of the usage of `onBlur` or `onKeyDown` is taken into consideration.
If `onBlur` or `onKeyDown` is set for some specific usage, the fix doesn't update the `value` prop to make the setting of `onBlur` and `onKeyDown` functional.
@kswang1029, please make e2e test for this commit and let me know if there's still some regression for some cases. Thank you very much.

I-Chenn committed:
Merge branch 'dev' into icchen/1906_annoyingTextInputInVectorDialog_NEW

kswang1029 commented:
based on [65e32d4](https://github.com/CARTAvis/carta-frontend/pull/2112/commits/65e32d40657fbd175da3c6a5f6b9f74f31edd068)
there are some e2e test failures
1. the clip min and clip max text fields in the render config widget are not set correctly based on the histogram data. They are currently initialized as 0 and 1.
![Screen Shot 2023-03-07 at 14 46 59](https://user-images.githubusercontent.com/20819712/223346686-a1f1b242-27d3-4fab-8447-306451d377d9.png)
2. the displayed value of the compoments spinbox is not set correctly after automatically finding number of components.
![Screen Shot 2023-03-07 at 14 49 17](https://user-images.githubusercontent.com/20819712/223346922-5c21ba1d-d5d4-4c2b-a1ad-f23afe821fe0.png)
3. (as a note only) the velocity/redshift box in the spectral line query widget behaves a bit differently to the key press actions. The corresponding e2e tests can have a fix for this new behavior.

I-Chenn committed:
Check if value is auto-updated

I-Chenn committed:
Adjust format

I-Chenn commented:
@kswang1029,

The fix in 6f682cc and b9abce4 now deal three different input actions that update the value showing in the input field if the input field is built with `SafeNumericInput`:
1. User's type-in (controlled by `safeHandleValueChanged`)
2. Updated by `onBlur` and `onKeydown` if these two props are set (controlled by on `onKeyDown` & `onKeyDown`)
3. Automatically updated by other function (controlled by the `reaction` in the `constructor`)

The result is shown below:

1. Render config widget problem
This problem is due to the fix in f2a9c6a (`onBlur` and `onKeydown` ) does not work properly. It is now fixed.

[renderConfig.webm](https://user-images.githubusercontent.com/106953609/223930990-300e147f-063b-49e4-9e50-afe0f467b92d.webm)

2. Spinbox is not set correctly:
This problem is due to we didn't take the automatic update into consideration, it is now fixed with adding reaction in the constructor in `SafeNumericInput`

[spinboxSetting.webm](https://user-images.githubusercontent.com/106953609/223932378-90bea600-039a-4e20-a56d-15a62d3b931d.webm)

@kswang1029 , Please make the e2e test again and see if there are still regressions. Sorry for making you test so many times.

kswang1029 commented:
@I-Chenn ~please merge dev otherwise e2e tests cannot be run due to text style unification.~ done

kswang1029 committed:
merge dev

I-Chenn commented:
> @I-Chenn ~please merge dev otherwise e2e tests cannot be run due to text style unification.~ done

Thanks for merging.

kswang1029 reviewed (approved):
looks good. please proceed code review.

I-Chenn committed:
Update changelog

YuHsuan-Hwang reviewed (approved):
The changes look good.

YuHsuan-Hwang commented on CHANGELOG.md:
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Added a toggle button to let users decide whether or not to keep the previously generated moment images ([#2054](https://github.com/CARTAvis/carta-frontend/issues/2054)).
 * Added settings in the image view settings widget for panning and zooming the images ([#1176](https://github.com/CARTAvis/carta-frontend/issues/1176)).
 ### Fixed
+* Fixed the issue of annoying text input fields in the vector rendering dialog ([#1906](https://github.com/CARTAvis/carta-frontend/issues/1906)).
```
@I-Chenn I think we could remove "in the vector rendering dialog" because we fixed all the inputs. Please also resolve the merge conflict, thanks.

I-Chenn replied:
Done in ac99a21.

I-Chenn committed:
Merge branch 'dev' into icchen/1906_annoyingTextInputInVectorDialog_NEW

I-Chenn committed:
Modify changelog
